apps/omg_watcher_info/lib/omg_watcher_info/db/repo.ex

```diff
@@ -32,17 +32,29 @@ defmodule OMG.WatcherInfo.DB.Repo do
 
   def insert_all_chunked(_schema_or_source, [], _opts), do: :ok
 
+  # Note that while this function allows us to get around postgres' max parameters limit
+  # by breaking large single INSERT statements into smaller ones, it also creates a bottleneck
+  # where out-of-memory errors could become a problem. See this issue for more information:
+  # https://github.com/omisego/elixir-omg/issues/1357
   def insert_all_chunked(schema_or_source, entries, opts) do
-    chunk_size = @max_params_count |> div(entries |> hd |> fields_count)
-
     utc_now = DateTime.utc_now()
+    entries = Enum.map(entries, fn entry -> Map.merge(entry, %{inserted_at: utc_now, updated_at: utc_now}) end)
+
+    chunk_size = chunk_size(entries |> hd)
 
     entries
-    |> Enum.map(fn entry -> Map.merge(entry, %{inserted_at: utc_now, updated_at: utc_now}) end)
     |> Stream.chunk_every(chunk_size)
     |> Enum.each(&insert_all(schema_or_source, &1, opts))
   end
 
+  # Note: an entry with 0 fields will cause a divide-by-zero error.
+  #
+  # DB.Repo.chunk_size(%{}) ==> (ArithmeticError) bad argument in arithmetic expression
+  #
+  # But we could not think of a case where this code happen, so no defensive
+  # checks here.
+  def chunk_size(entry), do: div(@max_params_count, fields_count(entry))
+
   defp fields_count(map) when is_map(map), do: map |> Kernel.map_size()
```

Comment:
```suggestion
  defp fields_count(map) when is_map(map), do: Kernel.map_size(map)
```

Reply:
question @jrhite why do we even need this? when are we inserting a `single map` and would worry it has to many elements?

Reply:
Not sure I understand your question? I didn't write the original code and to avoid another unexpected regression tried to limit the changes as much as possible.

Number of fields is needed to calculate `chunk_size`.